android/quest/src/main/java/org/smartregister/fhircore/quest/data/patient/PatientRepository.kt

```diff
@@ -243,16 +258,31 @@ constructor(
   }
 
   suspend fun getQuestionnaire(questionnaireResponse: QuestionnaireResponse): Questionnaire {
-    return if (questionnaireResponse.questionnaire != null) {
-      val questionnaireId = questionnaireResponse.questionnaire.split("/")[1]
-      loadQuestionnaire(questionnaireId = questionnaireId)
-    } else {
-      Timber.e(
-        Exception(
-          "Cannot open QuestionnaireResponse because QuestionnaireResponse.questionnaire is null"
+    return when {
+      questionnaireResponse.questionnaire.isNullOrBlank() -> {
+        Timber.e(
+          Exception(
+            "Cannot open QuestionnaireResponse because QuestionnaireResponse.questionnaire is null"
+          )
         )
-      )
-      Questionnaire()
+        Questionnaire()
+      }
+      else -> {
+        val questionnaireUrlList = questionnaireResponse.questionnaire.split("/")
```

Comment:
val questionnaireId = questionnaireResponse.questionnaire.split("/").last()

Reply:
I think, we also have few examples of data like this `"questionnaire": "Questionnaire/3440/_history/20"` and in that case the last splitted item would be 20 and it's not correct. I am not sure if it's a data issue. @maimoonak

Reply:
Ok. Got it. Right